### PR TITLE
Fix/clippy

### PR DESCRIPTION
**Pull request by yubiuser**
Fixes the clippy warning that occur with v1.78. being the new stable. This made the test fail (e.g [here](https://github.com/librespot-org/librespot/actions/runs/9043092501/job/24850355483)).
Addressed the following warnings:
- https://rust-lang.github.io/rust-clippy/master/index.html#/to_string_trait_impl
- https://rust-lang.github.io/rust-clippy/master/index.html#/assigning_clones
- https://rust-lang.github.io/rust-clippy/master/index.html#/incompatible_msrv
- https://rust-lang.github.io/rust-clippy/master/index.html#/derivable_impls

Sidenote: set the environment variables `RUST_BACKTRACE` and `RUSTFLAGS` in the devcontainer the same as they are in the github action workflow file.

### CODE REVIEW

File: core/src/mercury/types.rs [hunk 1]
```diff
@@ -49,15 +49,14 @@ impl From<MercuryError> for Error {
...
-impl ToString for MercuryMethod {
-    fn to_string(&self) -> String {
+impl std::fmt::Display for MercuryMethod {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match *self {
-            MercuryMethod::Get => "GET",
-            MercuryMethod::Sub => "SUB",
-            MercuryMethod::Unsub => "UNSUB",
-            MercuryMethod::Send => "SEND",
+            MercuryMethod::Get => write!(f, "GET"),
```
Comment (roderickvd) [inline core/src/mercury/types.rs:55]: Maybe do something like:
```rust
let s = match *self {
    MercuryMethod::Foo => "BAR",
}
write!(f, s)
```

File: core/src/session.rs [hunk 2]
```diff
@@ -358,7 +358,7 @@ impl Session {
                 loop {
                     match reader.read_event_into(&mut buf) {
                         Ok(Event::Start(ref element)) => {
-                            current_element = std::str::from_utf8(element)?.to_owned()
+                            std::str::from_utf8(element)?.clone_into(&mut current_element)
```
Comment (roderickvd) [inline core/src/session.rs:361]: Haven't done Rust development in a while, but oh wow, is this the new idiom? Doesn't make for a whole lot better reading... (not a review request; just a rant 😄)

**Review by roderickvd — COMMENTED** [review]
Thanks! Small request do keep it DRY.

File: core/src/mercury/types.rs (hunk 1 above)
Comment (yubiuser) [reply core/src/mercury/types.rs]: I pushed a commit with the change.

**Comment by roderickvd** [comment]
Perfect, thanks!

File: core/src/session.rs (hunk 2 above)
Comment (yubiuser) [reply core/src/session.rs:361]: Yeah, it comes from the clippy linting requested by https://rust-lang.github.io/rust-clippy/master/index.html#/assigning_clones

But I agree, it's harder to read (and others think so too - https://github.com/rust-lang/rust-clippy/issues/12778)

**Pull request merged**